Formula/lux.rb

```diff
@@ -0,0 +1,37 @@
+class Lux < Formula
+  desc "Lux (LUcid eXpect scripting) simplifies test automation and provides an Expect-style execution of commands"
+  homepage "https://github.com/hawk/lux"
+  url "https://github.com/hawk/lux/archive/lux-1.19.3.tar.gz"
+  sha256 "e2dd36275c76a087a4b4577bd056a6928a5e5ba5b00d8c366e2a4f94b16f921e"
+
+  head do
+    url "https://github.com/hawk/lux.git"
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
+  option "without-runpty", "Omit pseudo TTY wrapper"
+
+  depends_on "erlang"
+  depends_on "wxmac" => :optional
+
+  def install
+    ENV.deparallelize  # if your formula fails when building in parallel
+
+    system "autoconf"
+    system "./configure", "--prefix=#{prefix}/lux"
+    system "make"
+
+    #system "make", "install"
+    system "mkdir -p #{prefix}/lux"
+    system "tar cf - * | (cd #{prefix}/lux && tar xf -)"
+    system "mkdir -p #{prefix}/bin"
+    system "ln -s #{prefix}/lux/bin/lux #{prefix}/bin/lux"
+  end
+
+  test do
+    system "#{bin}/lux", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the feedback. I should not look to much to other formulas then. ;-)

I have a big test suite which can be used. But that is probably overkill. I will use a simple test, but more than printing the version. Even though the version test actually ensures that quite much of the app is working.